controllers/discovery/searchUser.js

```diff
@@ -80,8 +80,8 @@ const SearchUser = async (req, res) => {
           AND u.is_banned = false 
           AND u.verified_status != 'UNVERIFIED') 
       ORDER BY  
+          is_followed DESC,
           u.karma_score DESC,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the sorting order in the SQL query seem appropriate. The `is_followed` field is now sorted in descending order before `karma_score`, which means that followed users will be prioritized over non-followed ones with the same karma score. This change should improve the relevance of search results for users.
```diff
-          u.karma_score DESC,
-          is_followed ASC,
+          is_followed DESC,
+          u.karma_score DESC,
```
This change seems to align with the PR's goal of enhancing user discovery and search functionality by considering follower counts, follow status, and user preferences for anonymous direct messages. However, please ensure that this change doesn't conflict with any other requirements or expectations for the search results ordering.


<!-- This is an auto-generated comment by OSS CodeRabbit -->